wemake_python_styleguide/violations/best_practices.py

```diff
@@ -615,3 +617,23 @@ class RedundantForElseViolation(ASTViolation):
     #: Error message shown to the user.
     error_template = 'Found `else` in `for` loop with `break`'
     code = 436
+
+
+class RedundantFinallyViolation(ASTViolation):
+    """
+    Forbids to use ``finally`` in ``try`` block without ``except``.
+
+    Reasoning:
+        This rule will reduce complexity and improve readability.
+
+    Solution:
+        Refactor your ``try`` logic.
+
```

Comment:
You are also missing `Example::` section.
Using `open()` is a good use case to cover.